### PR TITLE
Ant, Maven and Gradle help texts

### PR DESCRIPTION
**Pull request by prawalagarwal**
Description: Java Tasks (Ant, Maven and Gradle) lack information as help texts on some fields inside the Code Coverage sub area. For example it is not clear that if the source directories path is relative to the repo root or the build file location.

Solution: Help texts have been modified to indicate the required information.

### CODE REVIEW

File: Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -23,11 +23,11 @@
...
   "loc.input.label.classFilesDirectories": "Class Files Directories",
-  "loc.input.help.classFilesDirectories": "Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html.",
+  "loc.input.help.classFilesDirectories": "Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. This path is relative to the location of ANT build.xml directory. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html for Jacoco and https://github.com/cobertura/cobertura/wiki/Ant-Task-Reference for Cobertura.",
```
Comment (AshwiniChalla) [inline Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson]: Changes are missing here

File: Tasks/Maven/task.json [hunk 2]
```diff
@@ -115,7 +115,7 @@
             "defaultValue": "",
             "required": false,
             "groupName": "codeCoverage",
-            "helpMarkDown": "Comma separated list of filters to include or exclude classes from collecting code coverage. For example, +:com.*,+:org.*,-:my.app*.*. Refer http://www.eclemma.org/jacoco/trunk/doc/maven.html.",
+            "helpMarkDown": "Comma separated list of filters to include or exclude classes from collecting code coverage. For example, +:com.*,+:org.*,-:my.app*.*. Refer http://www.eclemma.org/jacoco/trunk/doc/maven.html for Jacoco.",
```
Comment (AshwiniChalla) [inline Tasks/Maven/task.json]: Link for Cobertura?

File: Tasks/ANT/task.json [hunk 3]
```diff
@@ -110,7 +110,7 @@
             "defaultValue": ".",
             "required": true,
             "groupName": "codeCoverage",
-            "helpMarkDown": "Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html.",
+            "helpMarkDown": "Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. This path is relative to the location of ANT build.xml directory. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html for Jacoco and https://github.com/cobertura/cobertura/wiki/Ant-Task-Reference for Cobertura.",
```
Comment (AshwiniChalla) [inline Tasks/ANT/task.json]: Comma seperated list of relative paths from ant build file to directories containing class files.
How about this?

File: Tasks/Gradle/task.json [hunk 4]
```diff
@@ -127,7 +127,7 @@
             "defaultValue": "",
             "required": false,
             "groupName": "codeCoverage",
-            "helpMarkDown": "Comma separated list of filters to include or exclude classes from collecting code coverage. For example, +:com.*,+:org.*,-:my.app*.*. Refer https://docs.gradle.org/current/userguide/jacoco_plugin.html.",
```
Comment (AshwiniChalla) [inline Tasks/Gradle/task.json]: dont see the change of classFiles directories here

File: Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson [hunk 5]
```diff
@@ -24,9 +24,9 @@
   "loc.input.label.codeCoverageTool": "CodeCoverage Tool",
   "loc.input.help.codeCoverageTool": "Select the code coverage tool.",
   "loc.input.label.classFilesDirectories": "Class Files Directories",
-  "loc.input.help.classFilesDirectories": "Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. For example, classes1/Tests,classes2/Tests2. Refer https://docs.gradle.org/current/userguide/jacoco_plugin.html.",
```
Comment (AshwiniChalla) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: Is this not relative to gradlew?
If it is .. my suggestion is Comma seperated list of relative paths from Gradle Wrapper Script to directories containing class files

Comment (AshwiniChalla) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: Why are we removing the reference link?

File: Tasks/Maven/task.json [hunk 6]
```diff
@@ -124,7 +124,7 @@
             "label": "Class Files Directories",
             "required": false,
             "groupName": "codeCoverage",
-            "helpMarkDown": "This field is required for a multi module project. Code coverage reports are created using ANT task. Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html",
+            "helpMarkDown": "This field is required for a multi module project. Code coverage reports are created using ANT task. Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. This path is relative to the location of MAVEN pom.xml directory. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html",
```
Comment (AshwiniChalla) [inline Tasks/Maven/task.json]: Comma seperated list of relative paths from Maven POM file to directories containing class files. 

similar below.

File: Tasks/ANT/task.json (hunk 3 above)
Comment (AshwiniChalla) [inline Tasks/ANT/task.json]: Refer to the help text of Ant build file, Gradle Wrapper, Maven POM file fields of respective tasks for relative path

File: Tasks/ANT/task.json [hunk 7]
```diff
@@ -110,7 +110,7 @@
             "defaultValue": ".",
             "required": true,
             "groupName": "codeCoverage",
-            "helpMarkDown": "Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html.",
```
Comment (AshwiniChalla) [inline Tasks/ANT/task.json]: Let us give an example of target\classes,target\testClasses as this is usually the structure followed

File: Tasks/ANT/task.json [hunk 8]
```diff
@@ -120,7 +120,7 @@
             "defaultValue": "",
             "required": false,
             "groupName": "codeCoverage",
-            "helpMarkDown": "Comma separated list of filters to include or exclude classes from collecting code coverage. For example, +:com.*,+:org.*,-:my.app*.*. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html.",
+            "helpMarkDown": "Comma separated list of filters to include or exclude classes from collecting code coverage. For example, +:com.*,+:org.*,-:my.app*.*. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html for Jacoco and https://github.com/cobertura/cobertura/wiki/Ant-Task-Reference for Cobertura.",
```
Comment (AshwiniChalla) [inline Tasks/ANT/task.json]: in the example the * were not showing up in UI. Might be some string formatting issues

File: Tasks/Gradle/task.json (hunk 4 above)
Comment (prawalagarwal) [inline Tasks/Gradle/task.json]: Its not required here because this field is never visible.

File: Tasks/Maven/task.json (hunk 6 above)
Comment (prawalagarwal) [inline Tasks/Maven/task.json]: Done

File: Tasks/Maven/task.json (hunk 2 above)
Comment (prawalagarwal) [inline Tasks/Maven/task.json]: Added. This link although contains no useful information for this particular field.

File: Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson (hunk 5 above)
Comment (prawalagarwal) [inline Tasks/Gradle/Strings/resources.resjson/en-US/resources.resjson]: This source directories field is not visible and hence not modifying it now. Also I dont know why is the link showing up here. It does not exist in this text.

File: Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson [hunk 9]
```diff
@@ -23,11 +23,11 @@
   "loc.input.label.codeCoverageTool": "Code Coverage Tool",
   "loc.input.help.codeCoverageTool": "Select the code coverage tool. For JaCoCo, please make sure jacocoant.jar is available in lib folder of Ant installation. For cobertura, set up an environment variable COBERTURA_HOME pointing to the cobertura jars location.",
```
Comment (AshwiniChalla) [inline Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson]: remove please

File: Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson [hunk 10]
```diff
@@ -23,11 +23,11 @@
...
   "loc.input.label.classFilesDirectories": "Class Files Directories",
-  "loc.input.help.classFilesDirectories": "Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html.",
+  "loc.input.help.classFilesDirectories": "Comma separated list of realtive paths from ant build file to directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. For example, target/classes, target/testClasses. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html for Jacoco and https://github.com/cobertura/cobertura/wiki/Ant-Task-Reference for Cobertura.",
```
Comment (AshwiniChalla) [inline Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson]: relative spelling

File: Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson [hunk 11]
```diff
@@ -23,11 +23,11 @@
...
   "loc.input.label.classFilter": "Class Inclusion/Exclusion Filters",
-  "loc.input.help.classFilter": "Comma separated list of filters to include or exclude classes from collecting code coverage. For example, +:com.*,+:org.*,-:my.app*.*. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html.",
+  "loc.input.help.classFilter": "Comma separated list of filters to include or exclude classes from collecting code coverage. For example, +:com.*,+:org.*,-:my.app*.*. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html for Jacoco and https://github.com/cobertura/cobertura/wiki/Ant-Task-Reference for Cobertura.",
```
Comment (AshwiniChalla) [inline Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson]: * are still not visible

File: Tasks/Maven/task.json [hunk 12]
```diff
@@ -124,7 +124,7 @@
             "label": "Class Files Directories",
             "required": false,
             "groupName": "codeCoverage",
-            "helpMarkDown": "This field is required for a multi module project. Code coverage reports are created using ANT task. Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html",
```
Comment (AshwiniChalla) [inline Tasks/Maven/task.json]: Add a "specify" here and below in src file directories.
Specify comma seperated list of ....

**Comment by AshwiniChalla** [comment]
Approved with suggestions

File: Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson (hunk 9 above)
Comment (prawalagarwal) [inline Tasks/ANT/Strings/resources.resjson/en-US/resources.resjson]: Done

File: Tasks/ANT/task.json (hunk 7 above)
Comment (prawalagarwal) [inline Tasks/ANT/task.json]: Ok

File: Tasks/Maven/task.json (hunk 12 above)
Comment (prawalagarwal) [inline Tasks/Maven/task.json]: Done

**Pull request merged**